### PR TITLE
Fixed Drag & Drop issues

### LINKED ISSUE #9270: Drag and drop (iPadOS): The content is cloned but not moved
## 📝 Provide detailed reproduction steps (if any)

1.  Go to classic editor example.
2.  Drag and drop an image.

### ✔️ Expected result

The image moved.

### ❌ Actual result

The image was cloned.

https://user-images.githubusercontent.com/1099479/111492257-d10cda00-873c-11eb-9e45-2d38fc4173d1.mov

## 📃 Other details

*   Browser: Safari
*   OS: iPadOS 14.4.1

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

### LINKED ISSUE #9287: [Android] Editor remains focused on the selected link/comment/change.
## 📝 Provide detailed reproduction steps (if any)

1. Run Chrome and enter the responsive mode, select a device, for example, Galaxy S5
2. or open the Chrome browser on the Android device.
2. Open any editor's latest sample and select a part of the text with the link/comment/change
4. Try to set a caret in any other place of the editor.

It seems that it's related to the balloon toolbar - if some selection triggers the balloon, it can't be deselected.

### ✔️ Expected result

The caret should appear in the selected place.

### ❌ Actual result

Nothing happens, if the editor is focused, it always focuses the previously selected content.
error1

## 📃 Other details

* Browser: Chrome 89
* OS: Android 9
* First affected CKEditor version: **27.0.0**

### PR DESCRIPTION
**Pull request by niegowski**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Internal (clipboard): Drag and drop is disabled on Android. Closes #9287.

Internal (clipboard): Fixed setting drop effect while dragging in non-Gecko browsers. Closes #9270.

Internal (clipboard): Fixed crash while dragging removed element.

Internal (widget): Fixed selection jumping to the first table cell on selection handle click on Android.

---

### Additional information

### CODE REVIEW

File: packages/ckeditor5-clipboard/src/dragdrop.js [hunk 1]
```diff
@@ -278,11 +290,17 @@ export default class DragDrop extends Plugin {
...
 
 		// Dragging over the editable.
-		this.listenTo( viewDocument, 'dragenter', () => {
-			if ( editor.isReadOnly ) {
+		this.listenTo( viewDocument, 'dragenter', ( evt, data ) => {
+			if ( editor.isReadOnly || !this.isEnabled ) {
 				return;
 			}
 
+			if ( data.target ) {
```
Comment (Reinmar) [inline packages/ckeditor5-clipboard/src/dragdrop.js:298]:
```suggestion
			// Avoid the spellcheck re-scanning the content when the
			// drop marker moves through the content.
			// We revert this change on _finalizeDrag().
			if ( data.target ) {
```

File: packages/ckeditor5-clipboard/src/dragdrop.js [hunk 2]
```diff
@@ -278,11 +290,17 @@ export default class DragDrop extends Plugin {
...
+				view.change( writer => {
+					writer.setAttribute( 'spellcheck', 'false', data.target.root );
```
Comment (Reinmar) [inline packages/ckeditor5-clipboard/src/dragdrop.js:300]: If someone had this attribute set already, we should remember that.

**Comment by oleq** [comment]
We tested this branch in Android (emulator in Android Studio) and the basic caret movement on tap does not work.

**Comment by oleq** [comment]
Also, when tested in iOS we noticed that the editor loses focus on drop and sometimes the selection is stuck in a nested editable (when a drop target).

**Comment by oleq** [comment]
Also, something in this PR prevents Safari on iOS from moving selection between the root and the nested editable (image caption) on tap.

**Comment by oleq** [comment]
@ckeditor/qa-team Hey, can you check this out? This is the blocker before the release. 🙏

**Comment by FilipTokarski** [comment]
We've checked it and everything seems to be ok now 👍

**Pull request merged**